README.md

```diff
@@ -207,3 +196,40 @@ gradle uploadArchives
 
 Currently all builds are considered snapshots.  The archive name is based off of `git describe`.
 
+###How to contribute
+(Note: section inspired by, and some text copied from, [Apache Parquet](https://github.com/apache/parquet-mr))
+ 
+We welcome all contributions to the GATK project. The contribution can be a [issue report]( https://github.com/broadinstitute/gatk/issues) or a [pull request](https://github.com/broadinstitute/gatk/pulls). If you're not a commiter, you will need to [make a fork](https://help.github.com/articles/fork-a-repo/) of the gatk repository and [issue a pull request](https://help.github.com/articles/be-social/) from your fork.
+
+For ideas on what to contribute, check issues labeled ["Help wanted (Community)"](https://github.com/broadinstitute/gatk/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted+%28Community%29%22). Comment on the issue to indicate you're interested in contibuting code and for sharing your questions and ideas.
+
+To contribute a patch:
+* Break your work into small, single-purpose patches if possible. It’s much harder to merge in a large change with a lot of disjoint features.
+* Submit the patch as a GitHub pull request against the master branch. For a tutorial, see the GitHub guides on [forking a repo](https://help.github.com/articles/fork-a-repo/) and [sending a pull request](https://help.github.com/articles/be-social/). If applicable, include the issue number in the pull request name.
+* Make sure that your code passes all our tests. You can run the tests with `gradle test` in the root directory.
+* Add tests for all new code you've written. We prefer using tests but high quality integration tests that use small amounts of data are acceptable.
+* Follow the [**General guidelines for Hellbender developers**](https://github.com/broadinstitute/gatk#general-guidelines-for-hellbender-developers).
+
+We tend to do fairly close readings of pull requests, and you may get a lot of comments. Some things to consider:
+* Write tests for all new code.
+* Document all classes and public methods.
+* For all public methods, check validity of the arguments and throw `IllegalArgumentException` if invalid.
+* Use braces for control constructs, `if`, `for` etc.
+* Make classes, variables, parameters etc `final` unless there is a strong reason not to.
+* Give your operators some room. Not `a+b` but `a + b` and not `foo(int a,int b)` but `foo(int a, int b)`.
+* Generally speaking, stick to the [Google Java Style guide](http://google-styleguide.googlecode.com/svn/trunk/javaguide.html)
+
+Thank you for getting involved!
+
+##Discussions
+* [GATK forum](http://gatkforums.broadinstitute.org/)
+* [Issue tracker](https://github.com/broadinstitute/gatk/issues)
+* Discussions also take place in github pull requests
+* For committers, we have a publicly-visible google group [gatk-dev](https://groups.google.com/a/broadinstitute.org/forum/?hl=en#!forum/gatk-dev)
```

Comment:
For the record, I really don't like having a google group in addition to the forum and github. It's yet another completely separate venue for discussion; it's redundant and will cause difficulties down the road. Also, when it was created and shared it was not made clear that this group was public, and that discussions that need to be Broad-private should not happen on that list. In short, we should limit venues to github (for public things) and a private forum (which already exists) for private things that need to be on record (as opposed to hipchat for immediate transient discussions).